src/Events/Authorization/GenericCloudEventXacmlMapper.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections.Generic;
+using System.Security.Claims;
+
+using Altinn.Authorization.ABAC.Xacml.JsonProfile;
+using Altinn.Common.PEP.Constants;
+using Altinn.Common.PEP.Helpers;
+
+using CloudNative.CloudEvents;
+
+namespace Altinn.Platform.Events.Authorization
+{
+    /// <summary>
+    /// Utility class for converting Events to XACML request
+    /// </summary>
+    public static class GenericCloudEventXacmlMapper
+    {
+        /// <summary>
+        /// Create XACML request for multiple 
+        /// </summary>
+        /// <param name="user">The user</param>
+        /// <param name="events">The list of events</param>
+        public static XacmlJsonRequestRoot CreateMultiDecisionRequest(ClaimsPrincipal user, List<CloudEvent> events)
+        {
+            List<string> actionTypes = new() { "subscribe" };
+            var resourceCategory = CreateMultipleResourceCategory(events);
+            return CloudEventXacmlMapper.CreateMultiDecisionRequest(user, actionTypes, resourceCategory);
+        }
+
+        /// <summary>
+        /// Creates a multi resource category for the provided list of cloud events
+        /// </summary>
+        internal static List<XacmlJsonCategory> CreateMultipleResourceCategory(List<CloudEvent> events)
+        {
+            List<XacmlJsonCategory> resourcesCategories = new();
+            int counter = 1;
+
+            foreach (CloudEvent cloudEvent in events)
+            {
+                var resourceCategory = CreateResourceCategory(cloudEvent);
+                resourceCategory.Id = CloudEventXacmlMapper.ResourceId + counter.ToString();
```

Comment:
Would it look nice with a separator char here? Something that keeps the original ResourceId distinguisable.

Reply:
Resource id is simply an `R`, so the result will be R1, R2 for instance. I believe the category Id is easier to read without a seperator